### ISSUE TITLE
Can't create GKE cluster using vscode remote extensions

### ISSUE THREAD

**Issue by simonz130** [GoogleCloudPlatform/cloud-code-vscode#121]
**Environment**:
*(Hint: "Report Kubernetes Extension Issue on Github" command will fill these out for you.)*
Extension version:
VSCode version: 1.35.1
OS: Debian Stretch

**Description**:
I tried to use vscode remote extensions with cloud code. When trying to create GKE cluster in [remote container](https://code.visualstudio.com/docs/remote/containers), I get this error:
```shell
[Error - 00:53:19] (node:260) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/zeltser/.config/Code/User/settings.json'
[Error - 00:53:19] (node:260) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
[Error - 00:53:19] (node:260) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
[Error - 00:58:37] (node:260) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/zeltser/.config/Code/User/settings.json'
[Error - 00:58:37] (node:260) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
```
**Repro step**:
Install remote container. create a new app.
In .devcontainer, add this under `extensions` section:
```
{
	"extensions": [
		"GoogleCloudTools.cloudcode"
	]
}
```
Open the new app in remote container.
Now create a new GKE cluster.

**Comment by redborian** [comment]
Thanks for reporting this @simonz130. We haven't looked at remote extension scenario. We do have this in our roadmap. I will update this thread if anything comes up.

**Comment by bbangert** [comment]
I got something similar as well. In my case I was using the Remote SSH extension so that Cloud Code was running in the remote machine. The Kubernetes explorer worked fine as it installed and located kubectl fine for that. However, when trying to deploy, it said it couldn't find kubectl. This was pretty odd as the explorer was able to locate it, but however the deploy command works is not able to find it.

**Comment by hermes85pl** [comment]
I am experiencing this issue when trying to work in WSL 2 that is currently promoted to developers by Microsoft as groundbreaking (as it is in fact). To be honest, your extension is the only one from quite a few extensions that I use that does not support remote development yet. Please consider increasing the priority of this issue.
Also, I would like to add that having installed Google Cloud SDK under WSL 2 I would not like to be nagged to install it on my host if I happen to open my IDE locally (i.e. operating on host) instead of remotely (i.e. operating under WSL 2). Although you might consider this a separate issue, essentially this is part of the support for remote development, because this is exactly where this issue stems from: the diversity of ways of using the IDE with Microsoft's family of "remote development" extensions.

**Comment by kschaab** [comment]
This is fixed in Cloud Code 1.0.0
